src/diffusers/loaders.py

```diff
@@ -225,6 +230,19 @@ def load_attn_procs(self, pretrained_model_name_or_path_or_dict: Union[str, Dict
         is_custom_diffusion = any("custom_diffusion" in k for k in state_dict.keys())
 
         if is_lora:
+            is_new_lora_format = all(
+                key.startswith(self.unet_name) or key.startswith(self.text_encoder_name) for key in state_dict.keys()
+            )
+            if is_new_lora_format:
+                raise ValueError(
```

Comment:
I don't think it's a good idea that this fails. Imagine the following case:

A user does not have a pipeline but just a unet and wants to load the unet lora weights from a "text encoder + unet" serialization format. In this case the natural thing for me would be to extract the unet's lora, throw a warning that a "text encoder + unet" weight is loaded and that the text encoder is disregarded